lib/convert/widgets/dropdown_crypto.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../../wallet/model/crypto_list_view_data.dart';
+import '../../wallet/model/crypto_view_data.dart';
+import '../controller/providers.dart';
+
+class DropdownCrypto extends StatefulHookConsumerWidget {
+  final AsyncValue<CryptoListViewData> cryptoData;
+  final String type;
+  final String fromCrypto;
+
+  const DropdownCrypto({
+    Key? key,
+    required this.cryptoData,
+    required this.type,
+    required this.fromCrypto,
+  }) : super(key: key);
+
+  @override
+  ConsumerState<DropdownCrypto> createState() => _DropdownCryptoState();
+}
+
+class _DropdownCryptoState extends ConsumerState<DropdownCrypto> {
+  List<CryptoViewData> listCrypto = [];
+  CryptoViewData? dropdownValue;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+      for (CryptoViewData crypto
+          in widget.cryptoData.value!.cryptoViewDataList) {
+        listCrypto.add(crypto);
+      }
+
+      if (widget.type == "from") {
+        for (int index = 0; index <= listCrypto.length; index++) {
+          if (listCrypto[index].symbol == widget.fromCrypto) {
+            dropdownValue = listCrypto[index];
+            ref.watch(fromCryptoConvertAbrev.state).state =
+                listCrypto[index].symbol;
+            ref.watch(fromCryptoCotacaoProvider.state).state =
+                double.parse(listCrypto[index].currentPrice.toString());
+            break;
```

Comment:
Por quê você precisa de um break no if

Reply:
Esse break é usado para quebrar o ciclo do for que está acima da condição, dessa forma, quando são encontrados os valores necessários a execução é finalizada, sem ele, o ciclo continuaria e iriam acontecer comparações inúteis.